File: src/main/java/ru/serdtsev/homemoney/dto/CategoryStat.java
package ru.serdtsev.homemoney.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.math.BigDecimal;
import java.util.Objects;
import java.util.UUID;

@AllArgsConstructor
@Getter
@Setter
@ToString
public class CategoryStat {
  private UUID id;
  private UUID rootId;
  private String name;
  private BigDecimal amount;

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    CategoryStat that = (CategoryStat) o;
    return Objects.equals(id, that.id);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id);
  }
}

File: src/main/java/ru/serdtsev/homemoney/balancesheet/CategoryStat.kt
package ru.serdtsev.homemoney.balancesheet

import java.math.BigDecimal
import java.util.*

data class CategoryStat(
    var id: UUID,
    var rootId: UUID?,
    var name: String,
    var amount: BigDecimal) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as CategoryStat?
        return this.id == that!!.id
    }

    override fun hashCode(): Int {
        return Objects.hash(this.id)
    }
}
